pkg/activator/handler/requestevent_handler.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2018 The Knative Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package handler
+
+import (
+	"net/http"
+
+	"github.com/knative/serving/pkg/activator"
+	"github.com/knative/serving/pkg/autoscaler"
+)
+
+// ReqEvent represents an incoming/finished request with a given key
+type ReqEvent struct {
+	Key       string
+	EventType ReqEventType
+}
+
+// ReqEventType specifies the type of event (In/Out)
+type ReqEventType int
+
+const (
+	// ReqIn represents an incoming request
+	ReqIn ReqEventType = iota
+	// ReqOut represents a finished request
+	ReqOut
+)
```

Comment:
I intend to fold the event-types together with the event types of the queue-proxy since they resemble each other greatly. Didn't want to get hung up in refactoring as for this PR though.